src/map/script.c

```diff
@@ -7455,34 +7455,127 @@ BUILDIN(makeitem)
 			nameid = UNKNOWN_ITEM_ID;
 	} else {
 		nameid = script_getnum(st, 2);
-		if( nameid <= 0 || !itemdb->exists(nameid)) {
+		if (nameid <= 0 || !itemdb->exists(nameid)) {
 			ShowError("makeitem: Nonexistant item %d requested.\n", nameid);
+			script_pushint(st, 0);
 			return false; //No item created.
 		}
 	}
-	amount  = script_getnum(st,3);
-	mapname = script_getstr(st,4);
-	x       = script_getnum(st,5);
-	y       = script_getnum(st,6);
+	amount = script_getnum(st, 3);
+	mapname = script_getstr(st, 4);
+	x = script_getnum(st, 5);
+	y = script_getnum(st, 6);
 
-	if(strcmp(mapname,"this")==0) {
+	if (strcmp(mapname, "this") == 0) {
 		TBL_PC *sd;
 		sd = script->rid2sd(st);
-		if (!sd) return true; //Failed...
-		m=sd->bl.m;
-	} else
-		m=map->mapname2mapid(mapname);
+		if (sd == NULL) {
+			script_pushint(st, 0);
+			return false;
```

Comment:
if sd is NULL, it means that `script->rid2sd()` failed. rid2sd already reports the failure source (so you should return true here, if you don't want it to be reported twice). No need to push anything either, since `script->rid2sd()` already aborts the script on failure, so no return value will be examined.